server/parameters/util/YAMLParser.java

```diff
@@ -64,20 +69,27 @@ String description() {
 
         public abstract Help help(String path);
 
-        public static class Predefined<TYPE> extends KeyValue {
-
+        public static abstract class Static extends KeyValue {
```

Comment:
We already have `Dynamic` `KeyValue`s with dynamic keys, now I added `Static` `KeyValue`s with static keys (there is a strict set of acceptable keys, we can't introduce new keys without modifying the code). 

`Static` has two subclasses:
* `Predefined` (required for declaration) - already existed;
* `Optional` (not required for declaration) - a new one for `development-mode`.

Reply:
I think the use of `static` makes sense, good catch.